pkg/httpproxy/proxy.go

```diff
@@ -230,16 +234,27 @@ func (p *proxy) secretGetter(req *http.Request, cAuth string) SecretGetter {
 }
 
 func (p *proxy) checkCluster(req *http.Request, user user.Info, clusterID, credID string) (authorizer.Decision, error) {
-	cluster, err := p.clusters.Controller().Lister().Get("", clusterID)
+	cluster, err := p.mgmtClusters.Controller().Lister().Get("", clusterID)
 	if err != nil {
 		return authorizer.DecisionDeny, err
 	}
-	if cluster.Spec.EKSConfig == nil {
+
+	administratedByProvV2Cluster := imported.IsAdministratedByProvisioningCluster(cluster)
+	if (cluster.Spec.EKSConfig == nil || cluster.Spec.EKSConfig.AmazonCredentialSecret != credID) && !administratedByProvV2Cluster {
```

Comment:
I added a comment. There is some subtlety here because KEv2 clusters and provisioning v2 clusters are the only clusters that have a cloud credential on them. We don't check GKE and AKS clusters here because they have their own proxy.